scripts/pylib/twister/twisterlib.py

```diff
@@ -2719,6 +2721,15 @@ def run(self):
 
         sys.stdout.flush()
 
+    def gather_footprints(self, instance):
+        if self.suite.enable_size_report and not self.suite.cmake_only:
+            self.suite.calc_one_elf_size(instance)
```

Comment:
can this be moved into the same class? No need for it to be in TestSuite anymore. Maybe fold  `calc_one_elf_size` into this function directly.

Reply:
I moved calc_one_elf_size to ProjectBuilder class. However, I think it is better not to fold this into "gather_metrics". If at some point we decide to add other metrices it might be easier to maintain separate methods for separate metrices and just add more calls for such methods in gather_metrices.